doc/CONFIGURATION.md

```diff
@@ -141,3 +149,453 @@ node:
+
+One thing thing to note is the section of the `node` configuration called `storage`. This is really just an aliased/renamed `datastore` section that is used for the storing all our data. By this we mean things like files, documents, movies, etc... It is for all intents and purposes our "main storage".  The one exception to this is the example configuration file earlier in the documentation which enables the `countedStore` which is a reference counted blockstore. If this is used in other `datastore` configuration sections it is simply ignored.
+
+The current datastore types are supported:
+
+* pebble
+  * currently experimental, but is generally the fastest datastore next to pure in-memory stores
+  * it is generally okay, however it can cause unexpected errors, and should be used at your own risk.
+  * not quite as expensive on memory as badger
+* badger
+  * not quite as fast as pebble, but still extremely fast and stable.
+  * The badger default was adjusted to use `fileLoadingMode` with `FileIO` to enable use on low-memory devices.
+  * If using on server environments, change `fileLoadingMode` to 2.
+* leveldb
+  * generally the lowest memory requirements, and suitable for low-memory devices.
+  * generally the slowest datastore
+* memory
+  * consumes the most memory out of everything since it resides entirely in-memory
+  * all data is lost on crash, or restart of service and should only be used for testing
+
+The current `opts` for all `datastore` sections are as follows. Note that whenever an `opt`ion is left out, the default is implied:
+
+
+| Name            | Values                 | Details                          |
+|-----------------|------------------------|----------------------------------|
+| fileLoadingMode | 0 (FileIO), 2 (MemoryMap)| Specifies how objects are loaded when using the **badger** datastore type. FileIO is for memory constrained devices. Default is 0 (FileIO) |
+| withSync | false, true | Specifies whether to synchronize writes to disk when using the **pebble** datastore type. Default is false.
+
+# TemporalX
+
+The `temporalx` section is used to configure the TemporalX gRPC server that facilitates interaction with our custom node. Additionally it enables things like tracing, and monitoring.
+
+<p align="left">
+  · <a href="#api"><strong>API</strong></a>
+  <br>
+  · <a href="#prometheus"><strong>Prometheus</strong></a>
+  <br>
+  · <a href="#profiling"><strong>Profiling</strong></a>
+</p>
+
+## API
+
+The `api` section is used to configure the gRPC API server. 
+
+Configuration Options:
+
+* `tracing` enables/disables jaeger tracing
+* `listen_address` is the address to listen on for gRPC connections
+* `listen_proto` is the protocol the gRPC endpoint is using
+* `tls.cert_file` is the location to a tls certificate
```

Comment:
is the path to the server's TLS certificate, saved as a PEM? file.

Reply:
pem file, i made a note of this.